src/main/kotlin/no/nav/syfo/Bootstrap.kt

```diff
@@ -578,21 +580,51 @@ fun findOperasjonstype(
             ?.lastOrNull()
             ?: return 1
 
-    return if (itfh.infotrygdForesp.sMhistorikk.status.kodeMelding == "04" ||
-            (typeSMinfo.periode.arbufoerTOM != null && (typeSMinfo.periode.arbufoerTOM..periode.periodeFOMDato).daysBetween() > 1)) {
-        1
-    } else if (typeSMinfo.periode.arbufoerTOM != null && periode.periodeFOMDato.isAfter(typeSMinfo.periode.arbufoerTOM) ||
-            (typeSMinfo.periode.arbufoerTOM != null && periode.periodeFOMDato.isEqual(typeSMinfo.periode.arbufoerTOM)) ||
-            (typeSMinfo.periode.arbufoerTOM == null && (typeSMinfo.periode.arbufoerFOM..periode.periodeFOMDato).daysBetween() > 1)) {
-        2
-    } else if (typeSMinfo.periode.arbufoerFOM == periode.periodeFOMDato || (typeSMinfo.periode.arbufoerFOM != null && typeSMinfo.periode.arbufoerFOM.isBefore(periode.periodeFOMDato))) {
+    return if (endringSykmelding(periode, itfh, typeSMinfo)) {
         3
+    } else if (paafolgendeSykmelding(periode, itfh, typeSMinfo)) {
+        2
+    } else if (forstegangsSykmelding(periode, itfh, typeSMinfo)) {
+        1
     } else {
         log.error("Could not determined operasjonstype {}", fields(loggingMeta))
         throw RuntimeException("Could not determined operasjonstype")
     }
 }
 
+fun forstegangsSykmelding(
+    periode: HelseOpplysningerArbeidsuforhet.Aktivitet.Periode,
+    itfh: InfotrygdForespAndHealthInformation,
+    typeSMinfo: TypeSMinfo
+): Boolean =
+        itfh.infotrygdForesp.sMhistorikk.status.kodeMelding == "04" ||
+        (typeSMinfo.periode.arbufoerTOM != null && (typeSMinfo.periode.arbufoerTOM..periode.periodeFOMDato).daysBetween() > 1) ||
+        (typeSMinfo.periode.arbufoerTOM != null && (typeSMinfo.periode.arbufoerTOM..periode.periodeFOMDato).daysBetween() < -1)
```

Comment:
Hva er det denne siste sjekken skal fange? Jeg leser det som at sykmeldingen er tidligere enn siste dag i siste sykmelding i infotrygd. Kan det ikke være en endring?

Reply:
Jeg tror det blir riktig siden vi sjekker for endring først, dvs i if'en lenger opp. Men, det betyr at rekkefølgen på if-sjekkene i koden lenger opp er vesentlig og det burde kanskje ha vært en kommentar på det (selv om jeg egentlig hater kommentarer i koden og mener koden i seg selv bør være dokumentasjon nok 😛). På den annen side spørs det om noen tør å røre disse sjekkene når de først virker uansett :)

Men jeg mener fortsatt at vi burde ha brukt absoluttverdi her, da slipper vi å duplisere sjekken for både pluss og minus 1 ;)